src/api/users/test/e2e/utils.js

```diff
@@ -0,0 +1,134 @@
+    headers: {
+      Authorization: `bearer ${createServiceToken()}`,
+      'Content-Type': 'application/json',
+    },
+  });
+
+  return response;
+};
+
+// takes an array of users and sequentially gets them all
+// not to be confused with getting users via pagination
+const getUsers = async (users) => Promise.all(users.map((user) => getUser(user)));
+
+const createUsers = (numberOfUsers, sorted = false) => {
+  let users = [...Array(numberOfUsers).keys()].map((index) => {
+    return {
+      firstName: `TelescopeUser${index}`,
+      lastName: `TelescopeUser${index}`,
+      email: `TelescopeUser${index}@email.com`,
+      hash: hash(`TelescopeUser${index}@email.com`),
+      displayName: `TelescopeUser${index} TelescopeUser${index}`,
+      isAdmin: false,
+      isFlagged: false,
+      feeds: [`https://dev.to/feed/TelescopeUser${index}`],
+      github: {
+        username: `TelescopeUser${index}`,
+        avatarUrl:
+          'https://avatars.githubusercontent.com/u/7242003?s=460&u=733c50a2f50ba297ed30f6b5921a511c2f43bfee&v=4',
+      },
+    };
+  });
+
+  if (sorted) {
+    users.sort((a, b) => {
+      if (a.hash > b.hash) return 1;
+      if (a.hash < b.hash) return -1;
+      return 0;
+    });
+  }
+
+  return users.map((user) => {
+    delete user.hash;
+    return user;
+  });
+};
+
+const postUser = async (user) => {
+  const response = await fetch(`${USERS_URL}/${hash(user.email)}`, {
+    method: 'post',
+    headers: {
+      Authorization: `bearer ${createServiceToken()}`,
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify(user),
+  });
+
+  return response;
+};
+
+const postUsers = async (users) =>
+  Promise.all(
+    users.map((user) =>
+      postUser(user)
+        .then((res) => {
+          // We should get a 201 (created), but if the user exists, a 409 (which is fine here)
+          if (!(res.status === 201 || res.status === 409)) {
+            throw new Error(`got unexpected status ${res.status}`);
+          }
+        })
+        .catch((err) => {
+          console.error('Unable to create user with Users service', { err });
+        })
+    )
+  );
+
+// Delete the Telescope users we created in the Users service.
+const putUser = async (user) => {
+  const response = fetch(`${USERS_URL}/${hash(user.email)}`, {
+    method: 'put',
+    headers: {
+      Authorization: `bearer ${createServiceToken()}`,
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify(user),
+  });
+
+  return response;
+};
+
+// Delete the Telescope users we created in the Users service.
+const deleteUser = async (user) => {
+  const response = fetch(`${USERS_URL}/${hash(user.email)}`, {
+    method: 'delete',
+    headers: {
+      Authorization: `bearer ${createServiceToken()}`,
+    },
+  });
+
+  return response;
+};
+
+// Delete the Telescope users we created in the Users service.
+const deleteUsers = async (users) => Promise.all(users.map((user) => deleteUser(user)));
```

Comment:
This doesn't need `async` for the same reason.  You're rightfully returning the promise directly vs. `awaiting` a response.